pkg/virt-handler/notify-server/info.go

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+package eventsserver
+
+import (
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/handler-launcher-com/notify/info"
+	notifyv1 "kubevirt.io/kubevirt/pkg/handler-launcher-com/notify/v1"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type InfoServer struct{}
+
+func (i InfoServer) Info(context.Context, *info.NotifyInfoRequest) (*info.NotifyInfoResponse, error) {
+
+	// since this is the first versioned version, we only support the current versions
+	// add older versions as soon as they are supported
+	return &info.NotifyInfoResponse{
+		SupportedNotifyVersions:      []string{notifyv1.NotifyVersion},
+		SupportedDomainVersions:      []string{api.DomainVersion},
+		SupportedK8SMetaAPIVersions:  []string{metav1.SchemeGroupVersion.String()},
+		SupportedK8SEventAPIVersions: []string{corev1.SchemeGroupVersion.String()},
```

Comment:
This sounds a little backwards, but I'm actually a little nervous about dynamically populating the versions we support here from constants.  The reason is it would be easy for someone to update one of these const values (like api.DomainVersion) and that would mean our api dropped support for a previous version without anyone being aware.

I'd rather everything broken if api.DomainVersion incremented until we were forced to make a virt-handler api decision

Reply:
> This sounds a little backwards, but I'm actually a little nervous about dynamically populating the versions we support here from constants. The reason is it would be easy for someone to update one of these const values (like api.DomainVersion) and that would mean our api dropped support for a previous version without anyone being aware.

Oh, very fair point. I think using constants on server side only, and strings on client side will not only prevent that, but even let tests break if you don't deal with new versions on client side.

> I'd rather everything broken if api.DomainVersion incremented until we were forced to make a virt-handler api decision

tbh, I don't understand this comment, can you explain please?

Reply:
> tbh, I don't understand this comment, can you explain please?

I want to make sure all unit and functional tests related to this api fail if api.DomainVersion is incremented.  This forces us to address the compatibility issue when DomainVersion is changed because all our tests will fail.

Reply:
got it, and this will happen already with the latest changes (use exiting  constants on server side, but manual values on client side)

Reply:
perfect, thanks!